### PR TITLE
[SPARK-10531] [CORE] AppId is set as AppName in status rest api

### PR DESCRIPTION
**Pull request by zjffdu**
Verify it manually.

### CODE REVIEW

**Comment by andrewor14** [comment]
add to whitelist

**Comment by andrewor14** [comment]
@vanzin

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -455,7 +455,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _ui =
       if (conf.getBoolean("spark.ui.enabled", true)) {
         Some(SparkUI.createLiveUI(this, _conf, listenerBus, _jobProgressListener,
-          _env.securityManager, appName, startTime = startTime))
+          _env.securityManager, applicationId, appName, startTime = startTime))
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: `applicationId` hasn't been initialized yet here, so you shouldn't need to provide it in the constructor. Just call `setAppId` later as you're already doing.

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 2]
```diff
@@ -516,6 +516,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _applicationId = _taskScheduler.applicationId()
     _applicationAttemptId = taskScheduler.applicationAttemptId()
     _conf.set("spark.app.id", _applicationId)
+    _ui.foreach(_.setAppId(_applicationId))
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: This is actually the only change that seems necessary to fix the problem. All the other changes could be reverted, I think.

**Comment by zjffdu** [comment]
Thanks for the review @vanzin 
For SparkUI, yes it is not necessary to put appId in constructor. But for history server, the appId is known when creating SparkUI. Although in history server the appId of SparkUI is never used for now. But for safety, I put appId in constructor in case appId will be used in the future.

**Comment by vanzin** [comment]
@zjffdu there's no point in adding more code if a single line change fixes the issue. Then both live UIs and history server UIs are always created the same way. Consistency = good.

**Comment by zjffdu** [comment]
Push another commit to revert the unecessary change. And I think the appName is not set correctly in FsHistoryProvider.scala & Master.scala (Fix it in this commit). @vanzin  Please help review.

**Comment by vanzin** [comment]
LGTM. At first the changes to the app name seem weird, since it was nice to have the extra information show up in the browser's title bar when looking at the history server or the master UI, but I see that it would make the information exposed in the API wrong.

retest this please

**Comment by vanzin** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala [hunk 3]
```diff
@@ -146,7 +146,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           val ui = {
             val conf = this.conf.clone()
             val appSecManager = new SecurityManager(conf)
-            SparkUI.createHistoryUI(conf, replayBus, appSecManager, appId,
+            SparkUI.createHistoryUI(conf, replayBus, appSecManager, "",
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]: instead of `""` this should be `appInfo.name`

Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]: (and then the call to `ui.setAppName` later is not needed.)

Comment (vanzin) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]: (and then maybe `SparkUI.setAppName` may not be needed at all!)

**Comment by vanzin** [comment]
@zjffdu could you take a look at whether `SparkUI.setAppName` is really needed at all?

**Comment by zjffdu** [comment]
Push another commit to remove SparkUI#setAppName, it is not used anywhere. I think the original purpose for setAppName is a workaround for including extra info on the html title. If it is for this purpose, we should expose another api rather than using appName. Another change in the patch is rename appInfo to appAttemptInfo. (There's another appInfo outside, two same appInfo may cause potential bugs)

**Comment by vanzin** [comment]
LGTM. Merging to master.

**Pull request closed without merging**